docs/MODDING.md

```diff
@@ -0,0 +1,191 @@
+
+```cpp
+const uint16_t dayTimeMin = 0;
+const uint16_t dayTimeMax = 0xFFFF;
+ImGui::SliderScalar("Time", ImGuiDataType_U16, &gSaveContext.dayTime, &dayTimeMin, &dayTimeMax);
+```
+
+So this tells us that `gSaveContext.dayTime` is what we're looking for. Let's now do a global search for this to see if we can find where it is updated. We find the following in `soh/src/code/z_kankyo.c` line 925:
+
+```cpp
+if (IS_DAY || gTimeIncrement >= 0x190) {
+    gSaveContext.dayTime += gTimeIncrement;
+} else {
+    gSaveContext.dayTime += gTimeIncrement * 2; // time moves twice as fast at night
+}
+```
+
+We can make a quick change to this code to verify this is indeed what we are looking for, lets multiply the the gTimeIncrement by 10:
+
+```diff
+if (IS_DAY || gTimeIncrement >= 0x190) {
+-    gSaveContext.dayTime += gTimeIncrement;
++    gSaveContext.dayTime += gTimeIncrement * 10;
+} else {
+-    gSaveContext.dayTime += gTimeIncrement * 2; // time moves twice as fast at night
++    gSaveContext.dayTime += gTimeIncrement * 2 * 10; // time moves twice as fast at night
+}
+```
+
+Rebuild the game and launch it, then load a save file. You should see that the time of day is now moving much faster. Terrific! While we could wrap this up and call it a day, we could make this user configurable by making a few more changes. I think a slider would be good for this, there's a slider in the cheat menu that we can use as a reference. Let's find it in `soh/soh/SohMenuBar.cpp` around line 1120:
+
+```cpp
+UIWidgets::EnhancementSliderFloat("Hookshot Reach Multiplier: %.1fx", "##gCheatHookshotReachMultiplier", "gCheatHookshotReachMultiplier", 1.0f, 5.0f, "", 1.0f, false);
+```
+
+The float values being passed in here are `minumum`, `maximum`, and `default` respectively. We'll make our minimum 0.2 to allow it to move slower, and our maximum 5.0 to allow it to move up to 5x faster. We'll also set the default to 1.0 so that it doesn't change the behavior by default. Copy this line and paste it below, then make the relevant changes:
+
+```cpp
+UIWidgets::EnhancementSliderFloat("Time Multiplier: %.1fx", "##gCheatTimeMultiplier", "gCheatTimeMultiplier", 0.2f, 5.0f, "", 1.0f, false);
+```
+
+Now we need to replace our hard coded values with the new variable. We can do this by replacing the `10` with a cvar call
+
+```diff
+if (IS_DAY || gTimeIncrement >= 0x190) {
+-    gSaveContext.dayTime += gTimeIncrement * 10;
++    gSaveContext.dayTime += gTimeIncrement * CVarGetFloat("gCheatTimeMultiplier", 1.0f);
+} else {
+-    gSaveContext.dayTime += gTimeIncrement * 2 * 10;
++    gSaveContext.dayTime += gTimeIncrement * 2 * CVarGetFloat("gCheatTimeMultiplier", 1.0f);
+}
+```
+
+After rebuilding and launching the game, you should now see a new slider in the cheat menu that allows you to change the speed of the day/night cycle. Nice!
+
```

Comment:
Add something about resources if what the reader wants to do isn't already covered.
```suggestion
If you're ever not sure about how something would be implemented, you can always look at external resources like the [Cloudmodding Wiki](https://wiki.cloudmodding.com/oot/Main_Page) to get a better perspective.
```